status/reporter_gitlab.go

```diff
@@ -157,14 +158,24 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 		if err != nil {
 			return fmt.Errorf("error while getting all commitStatuses for sha %s: %w", r.sha, err)
 		}
-		existingCommitStatus := r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
+		existingCommitStatus = r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
```

Comment:
We might want to get the existingCommitStatus outside of this condition (e.g. [here](https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR154)) so we also have it available for the new condition. 
That's because in the current code, we'd only fetch the existingCommitStatus in case the `glState == gitlab.Running`, while the [other (new) condition](https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR171) also includes `glState == gitlab.Pending`.
Another option might be to combine these two conditions into one condition, but I think that's optional.

Reply:
+1